### PR TITLE
OT-537 #295 Share from App to iOS

### PR DESCRIPTION
**Pull request by CihanOezkan**
**Link to the given issue**
#295 
https://jira.open-xchange.com/browse/OT-537

**Describe what the problem was / what the new feature is**
Sharing from App to iOS

**Describe your solution**
Added new Method channel to trigger the iOS UIActivityViewController to share files and text with other Apps.

### CODE REVIEW

File: ios/OX Coi/Application/AppDelegate.swift [hunk 1]
```diff
@@ -34,8 +36,35 @@ class AppDelegate: FlutterAppDelegate {
                 } else {
                     result(nil)
                 }
+            } else if call.method.contains(Method.Sharing.SendSharedData) {
+                self.shareFile(arguments: call.arguments as! Dictionary<String, String>)
```
Comment (phranck) [inline ios/OX Coi/Application/AppDelegate.swift]: If you declare `call.arguments` as `[String: String]`, then you can...

File: ios/OX Coi/Application/AppDelegate.swift [hunk 2]
```diff
@@ -34,8 +36,35 @@ class AppDelegate: FlutterAppDelegate {
...
+                result(nil)
             }
         }
     }
+    
+    private func shareFile(arguments: Dictionary<String, String>) {
+        let args: [String: String] = arguments
```
Comment (phranck) [inline ios/OX Coi/Application/AppDelegate.swift]: ...get rid of this superfluous declaration. 😊

File: ios/OX Coi/Application/AppDelegate.swift (hunk 1 above)
Comment (CihanOezkan) [reply ios/OX Coi/Application/AppDelegate.swift]: Done

File: ios/OX Coi/Application/AppDelegate.swift (hunk 2 above)
Comment (CihanOezkan) [reply ios/OX Coi/Application/AppDelegate.swift]: Done

File: ios/OX Coi/Application/AppDelegate.swift (hunk 1 above)
Comment (phranck) [reply ios/OX Coi/Application/AppDelegate.swift]: Just one thing I just was thinking about...  
Let's assume `call.arguments` isn't a dictionary or it is even `nil`. What will happen if we do a force cast here? Right, the app will crash. That's not what we like to expect. So I suggest to use a `guard let ...` instead to be on the safe side.

**Pull request merged**